### PR TITLE
Add/remove relayer implementation

### PR DESCRIPTION
**Pull request by iorveth**
Addresses issues 
https://github.com/nymlab/vectis/issues/7,
https://github.com/nymlab/vectis/issues/3

### CODE REVIEW

File: contracts/proxy/src/contract.rs [hunk 1]
```diff
@@ -171,6 +178,47 @@ pub fn execute_relay(
...
 }
 
+/// Add relayer to the relayers set
+pub fn execute_add_relayer(
+    deps: DepsMut,
+    info: MessageInfo,
+    relayer_addr: Addr,
+) -> Result<Response, ContractError> {
+    // Authorize user or multisig
+    authorize_user_or_multisig(deps.as_ref(), &info.sender)?;
```
Comment (whalelephant) [inline contracts/proxy/src/contract.rs:188]: Perhaps needs to be `authorize_user_or_guardians`?  
Regardless if guardian is multisig, the guardian role should be able to do this.

File: contracts/proxy/src/contract.rs [hunk 2]
```diff
@@ -234,14 +282,30 @@ pub fn authorize_guardian_or_multisig(
...
         Some(multisig_address) if multisig_address.eq(sender) => Ok(()),
-        Some(_) => Err(ContractError::IsNotMultisig {}),
-        // if multisig is not set, ensure caller address is guardian
+        // otherwise do guardian auth
         _ => ensure_is_guardian(deps, sender),
     }
 }
 
+/// Is used to authorize user or multisig contract
+pub fn authorize_user_or_multisig(deps: Deps, sender: &Addr) -> Result<(), ContractError> {
```
Comment (whalelephant) [inline contracts/proxy/src/contract.rs:293]: Related to comment above, perhaps we can have it as either `user` or `guardians` which could be single guardian or multisig.

**Review by whalelephant — CHANGES_REQUESTED** [review]
The rest looks good!

File: contracts/proxy/src/contract.rs (hunk 1 above)
Comment (iorveth) [reply contracts/proxy/src/contract.rs:188]: > Perhaps needs to be `authorize_user_or_guardians`? Regardless if guardian is multisig, the guardian role should be able to do this.

It makes sense, both guardian and multisig should able to do this.

File: contracts/proxy/src/contract.rs (hunk 2 above)
Comment (iorveth) [reply contracts/proxy/src/contract.rs]: Done here https://github.com/nymlab/vectis/pull/8/commits/4c9d4185dde859779451e4b062dff54bc9831673

**Pull request merged**